File: Basics/OOPS/JAVA/Card.java
abstract public class Card implements PaymentMethod{
    private String cardNumber;
    private String cardHolderName;

    public Card(String cardNumber, String cardHolderName) {
        this.cardNumber = cardNumber;
        this.cardHolderName = cardHolderName;
    }

    public String getCardNumber() {
        return cardNumber;
    }

    public String getCardHolderName() {
        return cardHolderName;
    }
}

File: Basics/OOPS/CPP/Card.cpp
#ifndef CARD_CPP
#define CARD_CPP

#include<iostream>
#include "PaymentMethod.cpp"
using namespace std;

class Card : public PaymentMethod{
   protected:
      int cardNumber;
      string cardHolderName;
      
      public: Card(const int &number, const string &name){
           this->cardNumber = number;
           this->cardHolderName = name;
      }

      public: int getCardNumber(){
           return cardNumber;
      }
      public: string getCardHolderName(){
           return cardHolderName;
      }
};

#endif